File: tests/generic/gen_cor_063/Input/balance_poller.js
const fs = require('fs')
const Web3 = require('web3')
const commander = require('commander')

function log(data) {
    let timestamp = new Date().toISOString();
    const entry = `${timestamp} ${data}\n`;
    fs.appendFileSync(options.log_file, entry, { flag: 'a' });
}

function get_balance() {
  web3.eth.getBalance(address,
    function(error, result) {
      if (error) {
        log(`Error returned is ${error}`)
      } else {
        log(`Account balance is ${result}`)
        log(`Starting to run the polling loop`)
        task();
      }
    }
  )
}

function task() {
  setTimeout(function() {
    contract.methods.balanceOf(address).call(
      function(error, result) {
        if (error) {
          log(`Error returned is ${error}`)
        } else {
          log(`New balance = ${result}`)
          task(contract, address);
        }
     })
  }, 2000);
}

commander
  .version('1.0.0', '-v, --version')
  .usage('[OPTIONS]...')
  .option('--network_http <value>', 'Http connection URL to the network')
  .option('--address <value>', 'Contract address')
  .option('--contract_abi <value>', 'Contract ABI file')
  .option('--polling_address <value>', 'Address for the account to poll')
  .option('--log_file <value>', 'The output file to write to')
  .parse(process.argv)

const options = commander.opts()
const web3 = new Web3(`${options.network_http}`)

var json = fs.readFileSync(`${options.contract_abi}`)
var abi = JSON.parse(json)
const contract = new web3.eth.Contract(abi, `${options.address}`)
const address = options.polling_address

get_balance()

File: tests/generic/gen_cor_062/Input/balance_poller.py
from web3 import Web3
import logging, time
import argparse, json, sys

logging.basicConfig(format='%(asctime)s %(message)s', stream=sys.stdout, level=logging.INFO)


if __name__ == "__main__":
    parser = argparse.ArgumentParser(prog='balance_poller')
    parser.add_argument('-u', '--network_http', help='Connection URL')
    parser.add_argument('-a', '--address', help='Address of the contract')
    parser.add_argument('-b', '--contract_abi', help='Abi of the contract')
    parser.add_argument('-p', '--polling_address', help='Address of account to poll')
    args = parser.parse_args()

    web3 = Web3(Web3.HTTPProvider(args.network_http))
    with open(args.contract_abi) as f:
        contract = web3.eth.contract(address=args.address, abi=json.load(f))

    logging.info('Account balance is %d', web3.eth.get_balance(args.polling_address))
    logging.info('Starting to run the polling loop')
    last_balance = 0
    while True:
        balance = contract.functions.balanceOf(args.polling_address).call({"gas":1000000})
        if balance > last_balance:
            last_balance = balance
            logging.info('New balance = %s', balance)
        time.sleep(2)


